File: game_client/Assets/Scripts/ClientProtocol/CommonEnum/EAiChatRoleType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace CommonEnum
{

/// <summary>
/// AI聊天角色类型
/// </summary>
public enum EAiChatRoleType {
	[InspectorName("SYSTEM - idx[0] - 系统")]
	SYSTEM, //0 ==== 系统
	[InspectorName("USER - idx[1] - 用户")]
	USER, //1 ==== 用户
}

public class EAiChatRoleTypeComparer : IEqualityComparer<EAiChatRoleType>{
	public bool Equals(EAiChatRoleType x, EAiChatRoleType y) { return x == y; }
	public int GetHashCode(EAiChatRoleType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 2;
}
}

File: game_server/ServerProtocol/src/CommonEnum/EAiChatRoleType.java
package CommonEnum;

/*********
 * AI聊天角色类型
 **/
public enum EAiChatRoleType {
	SYSTEM, //0 ==== 系统
	USER, //1 ==== 用户
	;
public static final EAiChatRoleType[]  EAiChatRoleType_Values = EAiChatRoleType.values();
public static final int EAiChatRoleType_Length = EAiChatRoleType_Values.length;
public static EAiChatRoleType EAiChatRoleType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EAiChatRoleType_Length){ return null; }
	return EAiChatRoleType_Values[_ivalue];
}
}

